src/stream.rs

```diff
@@ -136,19 +147,30 @@ impl OutputStreamBuilder {
         self
     }
 
+    /// Set a callback that will be called when an error occurs with the stream
+    pub fn with_error_callback<E>(mut self, callback: E) -> Self
+    where
+        E: FnMut(cpal::StreamError) + Send + 'static,
+    {
+        self.error_callback = Box::new(callback);
+        self
+    }
+
     /// Open output stream using parameters configured so far.
-    pub fn open_stream(&self) -> Result<OutputStream, StreamError> {
+    pub fn open_stream(self) -> Result<OutputStream, StreamError> {
         let device = self.device.as_ref().expect("output device specified");
-        OutputStream::open(device, &self.config)
+
+        OutputStream::open(device, &self.config, self.error_callback)
     }
 
     /// Try opening a new output stream with the builder's current stream configuration.
     /// Failing that attempt to open stream with other available configurations
     /// supported by the device.
     /// If all attempts fail returns initial error.
-    pub fn open_stream_or_fallback(&self) -> Result<OutputStream, StreamError> {
+    pub fn open_stream_or_fallback(self) -> Result<OutputStream, StreamError> {
         let device = self.device.as_ref().expect("output device specified");
-        OutputStream::open(device, &self.config).or_else(|err| {
+
+        OutputStream::open(device, &self.config, self.error_callback).or_else(|err| {
             for supported_config in supported_output_configs(device)? {
                 if let Ok(handle) = Self::default()
```

Comment:
Default callback is used here if the first attempt fails... So custom callback will not be used in subsequent attempts...

Reply:
Thanks - good catch, I've resolved this now!